### PR TITLE
feat(doctrine): boolean filter like laravel filters

### PR DESCRIPTION
**Pull request by vinceAmstoutz**
| Q             | A
| ------------- | ---
| Branch?       | `main`
| License       | MIT
| Doc PR        | _Coming soon_

Goal:

Using filters as we use Laravel filters. Example with Symfony:
```php
#[ApiResource]
#[GetCollection(
    parameters: [
        'active' => new QueryParameter(
            filter: new BooleanFilter(),
        ),

        // support also for filter aliasing
        'enabled' => new QueryParameter(
            filter: new BooleanFilter(),
            property: 'active',
        ),
    ],
)]
#[ORM\Entity]
class FilteredBooleanParameter {
    // ...
}
```

TODO:
- [x] Check swagger UI to see what it's like
- [x] Do the job for Doctrine ODM (MongoDM)
- [x] Fix PHPStan
- [x] Fix MongoDM tests

### CODE REVIEW

File: src/Doctrine/Orm/Extension/ParameterExtension.php [hunk 1]
```diff
@@ -30,17 +31,21 @@ final class ParameterExtension implements QueryCollectionExtensionInterface, Que
...
     /**
      * @param array<string, mixed> $context
      */
     private function applyFilter(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, ?Operation $operation = null, array $context = []): void
     {
+        $filter = null;
+
         foreach ($operation?->getParameters() ?? [] as $parameter) {
-            if (!($v = $parameter->getValue()) || $v instanceof ParameterNotFound) {
+            if (($v = $parameter->getValue()) === null || $v instanceof ParameterNotFound) {
```
Comment (soyuka) [inline src/Doctrine/Orm/Extension/ParameterExtension.php:48]: yoda style?

File: src/Doctrine/Orm/Extension/ParameterExtension.php [hunk 2]
```diff
@@ -49,9 +54,20 @@ private function applyFilter(QueryBuilder $queryBuilder, QueryNameGeneratorInter
...
+            if (\is_object($filterId)) {
+                $filter = $filterId;
+                $filter->setManagerRegistry($this->managerRegistry);
+                $filter->setProperties($values);
+            }
+
             if ($filter instanceof FilterInterface) {
-                $filter->apply($queryBuilder, $queryNameGenerator, $resourceClass, $operation, ['filters' => $values, 'parameter' => $parameter] + $context);
+                $filter->apply($queryBuilder, $queryNameGenerator, $resourceClass, $operation,
+                    array_merge(['filters' => $values, 'parameter' => $parameter], $context)
```
Comment (soyuka) [inline src/Doctrine/Orm/Extension/ParameterExtension.php:69]: why array_merge?

File: src/Doctrine/Orm/Extension/ParameterExtension.php [hunk 3]
```diff
@@ -49,9 +54,20 @@ private function applyFilter(QueryBuilder $queryBuilder, QueryNameGeneratorInter
                 continue;
             }
 
-            $filter = $this->filterLocator->has($filterId) ? $this->filterLocator->get($filterId) : null;
+            if (\is_string($filterId) && $this->filterLocator->has($filterId)) {
+                $filter = $this->filterLocator->has($filterId) ? $this->filterLocator->get($filterId) : null;
+            }
+
+            if (\is_object($filterId)) {
```
Comment (soyuka) [inline src/Doctrine/Orm/Extension/ParameterExtension.php:61]: elseif?

File: src/Hydra/Serializer/CollectionFiltersNormalizer.php [hunk 4]
```diff
@@ -189,12 +190,13 @@ private function getSearch(string $resourceClass, array $parts, array $filters,
...
 
-    /**
-     * Gets a filter with a backward compatibility.
-     */
-    private function getFilter(string $filterId): ?FilterInterface
+    private function resolveFilter(object|string|null $filterId): ?\ApiPlatform\Doctrine\Orm\Filter\FilterInterface
     {
-        if ($this->filterLocator && $this->filterLocator->has($filterId)) {
+        if (\is_object($filterId)) {
+            return $filterId instanceof \ApiPlatform\Doctrine\Orm\Filter\FilterInterface ? $filterId : null;
```
Comment (soyuka) [inline src/Hydra/Serializer/CollectionFiltersNormalizer.php:196]: we can't tight doctrine to hydra here, not sure why this condition is there

File: src/Serializer/SerializerFilterContextBuilder.php [hunk 5]
```diff
@@ -54,11 +54,25 @@ public function createFromRequest(Request $request, bool $normalization, ?array
...
             }
         }
 
         return $context;
     }
+
+    private function resolveFilter(object|string|null $filterId): ?\ApiPlatform\Doctrine\Orm\Filter\FilterInterface
+    {
+        if (\is_object($filterId)) {
+            return $filterId instanceof \ApiPlatform\Doctrine\Orm\Filter\FilterInterface ? $filterId : null;
```
Comment (soyuka) [inline src/Serializer/SerializerFilterContextBuilder.php:69]: this condition is wrong

File: src/Symfony/Bundle/Resources/config/doctrine_orm.xml [hunk 6]
```diff
@@ -150,6 +150,7 @@
 
         <service id="api_platform.doctrine.orm.extension.parameter_extension" class="ApiPlatform\Doctrine\Orm\Extension\ParameterExtension" public="false">
             <argument type="service" id="api_platform.filter_locator" />
+            <argument type="service" id="doctrine" />
```
Comment (soyuka) [inline src/Symfony/Bundle/Resources/config/doctrine_orm.xml:153]: this is missing on-invalid=null

File: src/Validator/Metadata/Resource/Factory/ParameterValidationResourceMetadataCollectionFactory.php [hunk 7]
```diff
@@ -52,15 +50,10 @@ public function create(string $resourceClass): ResourceMetadataCollection
...
                 $parameters = $operation->getParameters() ?? new Parameters();
-                foreach ($parameters as $key => $parameter) {
+                foreach ($operation->getParameters() ?? [] as $key => $parameter) {
                     $parameters->add($key, $this->addSchemaValidation($parameter));
                 }
 
-                // As we deprecate the parameter validator, we declare a parameter for each filter transfering validation to the new system
-                if ($operation->getFilters() && 0 === $parameters->count()) {
-                    $parameters = $this->addFilterValidation($operation);
-                }
```
Comment (soyuka) [inline src/Validator/Metadata/Resource/Factory/ParameterValidationResourceMetadataCollectionFactory.php:62]: why did you remove this?

File: src/Validator/Metadata/Resource/Factory/ParameterValidationResourceMetadataCollectionFactory.php [hunk 8]
```diff
@@ -52,15 +50,10 @@ public function create(string $resourceClass): ResourceMetadataCollection
 
             foreach ($operations as $operationName => $operation) {
                 $parameters = $operation->getParameters() ?? new Parameters();
-                foreach ($parameters as $key => $parameter) {
+                foreach ($operation->getParameters() ?? [] as $key => $parameter) {
```
Comment (soyuka) [inline src/Validator/Metadata/Resource/Factory/ParameterValidationResourceMetadataCollectionFactory.php:53]: why the change?

**Review by soyuka — CHANGES_REQUESTED** [review]

File: src/Doctrine/Orm/Extension/ParameterExtension.php (hunk 1 above)
Comment (vinceAmstoutz) [reply src/Doctrine/Orm/Extension/ParameterExtension.php:48]: Oops

File: src/Doctrine/Orm/Extension/ParameterExtension.php (hunk 2 above)
Comment (vinceAmstoutz) [reply src/Doctrine/Orm/Extension/ParameterExtension.php:69]: In my opinion `array_merge` is more explicit but after some research, I will rollback

File: src/Doctrine/Orm/Extension/ParameterExtension.php (hunk 3 above)
Comment (vinceAmstoutz) [reply src/Doctrine/Orm/Extension/ParameterExtension.php:61]: Here I've tried not to use the “else” keyword, according to Object Calisthenics "Don't Use The ELSE Keyword" principle. But I'm going to refactor it to look better.

**Comment by vinceAmstoutz** [comment]
@soyuka looks like this in Swagger UI:
image
image
image

**Comment by vinceAmstoutz** [comment]
@soyuka It's almost the same code between Orm and Odm but the tests only failed for Odm. Am I missing something?

File: src/Doctrine/Odm/Filter/AbstractFilter.php [hunk 9]
```diff
@@ -56,11 +60,16 @@ public function apply(Builder $aggregationBuilder, string $resourceClass, ?Opera
      */
     abstract protected function filterProperty(string $property, $value, Builder $aggregationBuilder, string $resourceClass, ?Operation $operation = null, array &$context = []): void;
 
-    protected function getManagerRegistry(): ManagerRegistry
+    protected function getManagerRegistry(): ?ManagerRegistry
     {
         return $this->managerRegistry;
     }
 
+    public function setManagerRegistry(?ManagerRegistry $managerRegistry): ?ManagerRegistry
```
Comment (soyuka) [inline src/Doctrine/Odm/Filter/AbstractFilter.php:68]: we need to add an interface for this and check that a filter implements the interface before calling `setManagerRegistry`

File: src/Doctrine/Odm/Filter/BooleanFilter.php [hunk 10]
```diff
@@ -139,4 +144,27 @@ protected function filterProperty(string $property, $value, Builder $aggregation
...
+        $key = $parameter->getKey();
+
+        return [
+            new OpenApiParameter(
+                name: $key,
+                in: $in,
+                required: false,
+                schema: ['type' => 'boolean'],
+            ),
+        ];
```
Comment (soyuka) [inline src/Doctrine/Odm/Filter/BooleanFilter.php:168]: in fact let's not implement this here. We need this only for filters like the DateFilter that adds keys like `date[before]=`

File: src/Doctrine/Orm/Extension/ParameterExtension.php [hunk 11]
```diff
@@ -71,4 +83,26 @@ public function applyToItem(QueryBuilder $queryBuilder, QueryNameGeneratorInterf
...
+     * @param $values array<string, string>
+     *
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
+     */
+    private function getFilter(AbstractFilter|string|null $filterId, array $values = []): ?FilterInterface
+    {
+        if ($filterId instanceof AbstractFilter) {
+            $filterId->setManagerRegistry($this->managerRegistry);
+            $filterId->setProperties($values);
```
Comment (soyuka) [inline src/Doctrine/Orm/Extension/ParameterExtension.php:97]: same as for the ODM

File: src/Doctrine/Orm/Filter/BooleanFilter.php [hunk 12]
```diff
@@ -145,4 +150,27 @@ protected function filterProperty(string $property, $value, QueryBuilder $queryB
...
+        $key = $parameter->getKey();
+
+        return [
+            new OpenApiParameter(
+                name: $key,
+                in: $in,
+                required: false,
+                schema: ['type' => 'boolean'],
+            ),
+        ];
```
Comment (soyuka) [inline src/Doctrine/Orm/Filter/BooleanFilter.php:174]: same as for the ODM filter

File: tests/Functional/Parameters/BooleanFilterTest.php [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+    /**
+     * @throws TransportExceptionInterface
+     * @throws ServerExceptionInterface
+     * @throws RedirectionExceptionInterface
+     * @throws DecodingExceptionInterface
+     * @throws ClientExceptionInterface
+     */
+    private function assertBooleanFilterResponse(?string $activeValue, int $expectedCount, string $param = 'active'): void
+    {
+        $route = $this->getRoute($activeValue, $param);
```
Comment (soyuka) [inline tests/Functional/Parameters/BooleanFilterTest.php:116]: let's avoid these kind of functions, you should rather use a phpunit data provider and use different values inside the test.

File: tests/Functional/Parameters/BooleanFilterTest.php [hunk 14]
```diff
@@ -0,0 +1,187 @@
...
+            $entity = new $resource(active: $activeValue);
+            $manager->persist($entity);
+        }
+
+        $manager->flush();
+    }
+
+    private function getRoute(?string $value, string $param = 'active'): string
+    {
+        return \sprintf('%s?%s=%s', self::ROUTE, $param, $value);
```
Comment (soyuka) [inline tests/Functional/Parameters/BooleanFilterTest.php:185]: not a fan of this as well, it makes tests too abstract, I'd like to know when reading a test how the route is, looking for the logic inside the test case is not convenient imo

**Review by soyuka — COMMENTED** [review]
Nice start! A few comments left to be done!

File: src/Doctrine/Odm/Extension/ParameterExtension.php [hunk 15]
```diff
@@ -72,4 +83,26 @@ public function applyToItem(Builder $aggregationBuilder, string $resourceClass,
...
+     * @param $values array<string, string>
+     *
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
+     */
+    public function getFilter(AbstractFilter|string|null $filterId, array $values = []): ?FilterInterface
+    {
+        if ($filterId instanceof AbstractFilter) {
+            $filterId->setManagerRegistry($this->managerRegistry);
+            $filterId->setProperties($values);
```
[the remark this answers is not in the document]
Comment (vinceAmstoutz) [reply src/Doctrine/Odm/Extension/ParameterExtension.php:97]: @soyuka `$parameter->getExtraProperties()` returns something like `[“_api_values” => “true”]`. It doesn't work with the current code if you set it to properties. However `$values` contains `[“active” => “true”]` which works.

File: src/Doctrine/Odm/Extension/ParameterExtension.php [hunk 16]
```diff
@@ -72,4 +83,26 @@ public function applyToItem(Builder $aggregationBuilder, string $resourceClass,
...
         $this->applyFilter($aggregationBuilder, $resourceClass, $operation, $context);
     }
+
+    /**
+     * @param $values array<string, string>
+     *
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
+     */
+    public function getFilter(AbstractFilter|string|null $filterId, array $values = []): ?FilterInterface
```
[the remark this answers is not in the document]
Comment (vinceAmstoutz) [reply src/Doctrine/Odm/Extension/ParameterExtension.php:93]: yep!

File: src/Doctrine/Odm/Extension/ParameterExtension.php [hunk 17]
```diff
@@ -45,7 +55,8 @@ private function applyFilter(Builder $aggregationBuilder, ?string $resourceClass
                 continue;
             }
 
-            $filter = $this->filterLocator->has($filterId) ? $this->filterLocator->get($filterId) : null;
+            $filter = $this->getFilter($filterId, $values);
```
[the remark this answers is not in the document]
Comment (vinceAmstoutz) [reply src/Doctrine/Odm/Extension/ParameterExtension.php:58]: Why? It extracts complexity

File: src/Doctrine/Odm/Filter/AbstractFilter.php [hunk 18]
```diff
@@ -56,11 +60,16 @@ public function apply(Builder $aggregationBuilder, string $resourceClass, ?Opera
      */
     abstract protected function filterProperty(string $property, $value, Builder $aggregationBuilder, string $resourceClass, ?Operation $operation = null, array &$context = []): void;
 
-    protected function getManagerRegistry(): ManagerRegistry
+    protected function getManagerRegistry(): ?ManagerRegistry
```
[the remark this answers is not in the document]
Comment (vinceAmstoutz) [reply src/Doctrine/Odm/Filter/AbstractFilter.php:63]: Possibly, but since you suggested making the `ManagerRegistry` nullable ([see code here](https://github.com/api-platform/core/blob/1892e6aeffbcc9c748dd0b3d929090f38f835274/src/Doctrine/Orm/Filter/AbstractFilter.php#L34)), I’ve simply updated the getter to maintain consistency. Otherwise, what happens if the `managerRegistry` property is null and we call `getManagerRegistry`?"

File: src/Doctrine/Odm/Filter/BooleanFilter.php (hunk 10 above)
Comment (vinceAmstoutz) [reply src/Doctrine/Odm/Filter/BooleanFilter.php:168]: What does that mean? In what way should I remove it? And just for Odm or also for Orm?

File: src/Doctrine/Orm/Filter/BooleanFilter.php (hunk 12 above)
Comment (vinceAmstoutz) [reply src/Doctrine/Orm/Filter/BooleanFilter.php:174]: What does that mean? In what way should I remove it?

File: tests/Functional/Parameters/BooleanFilterTest.php (hunk 13 above)
Comment (vinceAmstoutz) [reply tests/Functional/Parameters/BooleanFilterTest.php:116]: Refactored using a DataProvider with a Generator.

File: src/Doctrine/Odm/Filter/AbstractFilter.php (hunk 9 above)
Comment (vinceAmstoutz) [reply src/Doctrine/Odm/Filter/AbstractFilter.php:68]: Don't all filters implement `AbstractFilter.php`?

Comment (soyuka) [reply src/Doctrine/Odm/Filter/AbstractFilter.php:68]: no

File: src/Doctrine/Odm/Extension/ParameterExtension.php (hunk 17 above)
[the remark this answers is not in the document]
Comment (soyuka) [reply src/Doctrine/Odm/Extension/ParameterExtension.php:58]: because at some point we'll not use the filter locator anymore, and we'll have to rollback this function. I'd prefer to introduce a new code path  for this, as it'll simplify the maintenance later on. 
Adding a new function increases complexity imo as we need to read the function to understand what it does.

File: src/Doctrine/Odm/Extension/ParameterExtension.php (hunk 15 above)
[the remark this answers is not in the document]
Comment (soyuka) [reply src/Doctrine/Odm/Extension/ParameterExtension.php:97]: This is a wrong assumption, as values are not representing the properties, this should be one of: 

- `$parameter->getExtraProperties()['_properties']`
- `[$parameter->getProperty()]`
- `[$parameter->getKey()]`

For example when you alias a filter to a property values will contain [key => value] although it'd filter on a given property.

[the remark this answers is not in the document]
Comment (vinceAmstoutz) [reply src/Doctrine/Odm/Extension/ParameterExtension.php:97]: Ok, I understand but  I've tried all 3 solutions to pass them to `setProperties()` and nothing works. I think I need an array with the parameter key (active) in the index and the filter value (true) in value, because that's the only solution that works actually, and I see something like that also in the following code, which then feeds `$values` if we check usages
image
image

File: src/Doctrine/Odm/Filter/AbstractFilter.php (hunk 9 above)
Comment (vinceAmstoutz) [reply src/Doctrine/Odm/Filter/AbstractFilter.php:68]: I created the `ManagerRegistryConfigurableInterface` interface, implemented the interface in `AbstractFilter` and added a tester in `ParameterExtension` for `Odm` + `ORM`. What do you think?

File: src/Doctrine/Odm/Extension/ParameterExtension.php (hunk 17 above)
[the remark this answers is not in the document]
Comment (vinceAmstoutz) [reply src/Doctrine/Odm/Extension/ParameterExtension.php:58]: Done

File: src/Doctrine/Odm/Filter/AbstractFilter.php (hunk 18 above)
[the remark this answers is not in the document]
Comment (vinceAmstoutz) [reply src/Doctrine/Odm/Filter/AbstractFilter.php:63]: Done

File: src/Doctrine/Orm/Extension/ParameterExtension.php (hunk 11 above)
Comment (vinceAmstoutz) [reply src/Doctrine/Orm/Extension/ParameterExtension.php:97]: https://github.com/api-platform/core/pull/6749#discussion_r1824635345

File: src/Doctrine/Odm/Extension/ParameterExtension.php [hunk 19]
```diff
@@ -45,14 +56,30 @@ private function applyFilter(Builder $aggregationBuilder, ?string $resourceClass
...
+
+            if (!($filter instanceof FilterInterface)) {
+                return;
+            }
+
+            if (!$filter->hasManagerRegistry() && $filter instanceof ManagerRegistryConfigurableInterface) {
+                $filter->setManagerRegistry($this->managerRegistry);
+            }
+
+            if ([] === $filter->getProperties() || null === $filter->getProperties()) {
```
Comment (soyuka) [inline src/Doctrine/Odm/Extension/ParameterExtension.php:73]:
```suggestion
            if (!$filter->getProperties()) {
```

File: src/Doctrine/Odm/Extension/ParameterExtension.php [hunk 20]
```diff
@@ -45,14 +56,30 @@ private function applyFilter(Builder $aggregationBuilder, ?string $resourceClass
...
+                $filterId instanceof AbstractFilter => $filterId,
+                \is_string($filterId) && $this->filterLocator->has($filterId) => $this->filterLocator->get($filterId),
+                default => null,
+            };
+
+            if (!($filter instanceof FilterInterface)) {
+                return;
+            }
+
+            if (!$filter->hasManagerRegistry() && $filter instanceof ManagerRegistryConfigurableInterface) {
```
Comment (soyuka) [inline src/Doctrine/Odm/Extension/ParameterExtension.php:69]:
```suggestion
            if ($filter instanceof ManagerRegistryConfigurableInterface && !$filter->hasManagerRegistry()) {
```

File: src/Doctrine/Odm/Extension/ParameterExtension.php [hunk 21]
```diff
@@ -45,14 +56,30 @@ private function applyFilter(Builder $aggregationBuilder, ?string $resourceClass
...
+                $filter->setManagerRegistry($this->managerRegistry);
+            }
+
+            if ([] === $filter->getProperties() || null === $filter->getProperties()) {
+                $key = $parameter->getProperty() ?? $parameter->getKey();
+                $filter->setProperties([$key => []]);
```
Comment (soyuka) [inline src/Doctrine/Odm/Extension/ParameterExtension.php:75]:
```suggestion
                $filter->setProperties([$parameter->getProperty() ?? $parameter->getKey() => []]);
```

File: src/Doctrine/Odm/Filter/AbstractFilter.php [hunk 22]
```diff
@@ -56,18 +60,35 @@ public function apply(Builder $aggregationBuilder, string $resourceClass, ?Opera
...
     {
+        if (!$this->hasManagerRegistry()) {
+            throw new \RuntimeException('ManagerRegistry must be initialized before accessing it.');
+        }
+
         return $this->managerRegistry;
     }
 
-    protected function getProperties(): ?array
+    public function setManagerRegistry(?ManagerRegistry $managerRegistry): ?ManagerRegistry
```
Comment (soyuka) [inline src/Doctrine/Odm/Filter/AbstractFilter.php:77]:
```suggestion
    public function setManagerRegistry(ManagerRegistry $managerRegistry): void
```

File: src/Doctrine/Odm/Filter/ManagerRegistryConfigurableInterface.php [hunk 23]
```diff
@@ -0,0 +1,21 @@
...
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Doctrine\Odm\Filter;
+
+use Doctrine\Bundle\MongoDBBundle\ManagerRegistry;
+
+interface ManagerRegistryConfigurableInterface
```
Comment (soyuka) [inline src/Doctrine/Odm/Filter/ManagerRegistryConfigurableInterface.php:18]:
```suggestion
interface ManagerRegistryAwareInterface
```

File: src/Doctrine/Odm/Filter/ManagerRegistryConfigurableInterface.php [hunk 24]
```diff
@@ -0,0 +1,21 @@
...
+{
+    public function setManagerRegistry(?ManagerRegistry $managerRegistry): ?ManagerRegistry;
```
Comment (soyuka) [inline src/Doctrine/Odm/Filter/ManagerRegistryConfigurableInterface.php:20]:
```suggestion
    public function setManagerRegistry(ManagerRegistry $managerRegistry): void;
```

Can you add `getManagerRegistry` and `hasManagerRegistry` to the interface?

File: src/Doctrine/Orm/Filter/ManagerRegistryConfigurableInterface.php [hunk 25]
```diff
@@ -0,0 +1,21 @@
...
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Doctrine\Orm\Filter;
+
+use Doctrine\Persistence\ManagerRegistry;
+
+interface ManagerRegistryConfigurableInterface
+{
+    public function setManagerRegistry(?ManagerRegistry $managerRegistry): ?ManagerRegistry;
```
Comment (soyuka) [inline src/Doctrine/Orm/Filter/ManagerRegistryConfigurableInterface.php:20]:
```suggestion
    public function setManagerRegistry(ManagerRegistry $managerRegistry): void

File: src/Doctrine/Orm/Filter/ManagerRegistryConfigurableInterface.php [hunk 26]
```diff
@@ -0,0 +1,21 @@
...
+}
```
Comment (soyuka) [inline src/Doctrine/Orm/Filter/ManagerRegistryConfigurableInterface.php:21]: can you add `has` and `get` methods to the interface?

File: src/Doctrine/Orm/Filter/ManagerRegistryConfigurableInterface.php [hunk 27]
```diff
@@ -0,0 +1,21 @@
...
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Doctrine\Orm\Filter;
+
+use Doctrine\Persistence\ManagerRegistry;
+
+interface ManagerRegistryConfigurableInterface
```
Comment (soyuka) [inline src/Doctrine/Orm/Filter/ManagerRegistryConfigurableInterface.php:18]:
```suggestion
interface ManagerRegistryAwareInterface
```

File: src/Doctrine/Orm/PropertyHelperTrait.php [hunk 28]
```diff
@@ -29,7 +29,7 @@
  */
 trait PropertyHelperTrait
 {
-    abstract protected function getManagerRegistry(): ManagerRegistry;
+    abstract protected function getManagerRegistry(): ?ManagerRegistry;
```
Comment (soyuka) [inline src/Doctrine/Orm/PropertyHelperTrait.php:32]: let's not change the return argument

File: tests/Functional/Parameters/BooleanFilterTest.php [hunk 29]
```diff
@@ -0,0 +1,173 @@
...
+     */
+    #[DataProvider('invalidBooleanFilterProvider')]
+    public function testBooleanFilterWithInvalidValues(string $url): void
+    {
+        $response = self::createClient()->request('GET', $url);
+
+        $this->assertEquals(422, $response->getStatusCode(), \sprintf('Expected status code 422 for URL %s', $url));
+        $this->assertJsonContains([
+            '@context' => '/contexts/ConstraintViolationList',
+            '@id' => '/validation_errors/c1051bb4-d103-4f74-8988-acbcafc7fdc3',
```
Comment (soyuka) [inline tests/Functional/Parameters/BooleanFilterTest.php:132]:
```suggestion
```

File: tests/Functional/Parameters/BooleanFilterTest.php [hunk 30]
```diff
@@ -0,0 +1,173 @@
...
+            '@type' => 'ConstraintViolationList',
+            'status' => 422,
+            'violations' => [
+                0 => [
+                    'propertyPath' => 'active',
+                    'message' => 'This value should not be blank.',
+                    'code' => 'c1051bb4-d103-4f74-8988-acbcafc7fdc3',
```
Comment (soyuka) [inline tests/Functional/Parameters/BooleanFilterTest.php:139]:
```suggestion
```

File: tests/Functional/Parameters/BooleanFilterTest.php [hunk 31]
```diff
@@ -0,0 +1,173 @@
...
+                ],
+            ],
+            'detail' => 'active: This value should not be blank.',
+            'description' => 'active: This value should not be blank.',
+            'type' => '/validation_errors/c1051bb4-d103-4f74-8988-acbcafc7fdc3',
```
Comment (soyuka) [inline tests/Functional/Parameters/BooleanFilterTest.php:144]:
```suggestion
```

**Review by soyuka — CHANGES_REQUESTED** [review]
